### PR TITLE
Add surface of trim chunks to scrub

### PR DESCRIPTION
**Pull request by kuenishi**
This pull request adds a gate to trim chunks. We had several discussions on "fill" API of CORFU but for now this is just for trim.

### CODE REVIEW

**Comment by jadeallenx** [comment]
Nice start Kota :cake:

**Comment by kuenishi** [comment]
@slfritchie Do you think is it OK to merge for now? To avoid conflict against Mark's #6, I'd like to touch machi_flu1 after this got merged.

File: src/machi.proto [hunk 1]
```diff
@@ -130,6 +130,7 @@ message Mpb_ErrorResp {
 // append_chunk()  : Mpb_AppendChunkReq and Mpb_AppendChunkResp
 // write_chunk()   : Mpb_WriteChunkReq and Mpb_WriteChunkResp
 // read_chunk()    : Mpb_ReadChunkReq and Mpb_ReadChunkResp
+// trim_chunk()    : Mpb_TrimCHunkReq and Mpb_TrimChunkResp
```
Comment (slfritchie) [inline src/machi.proto]: Please s/CHunk/Chunk/

File: src/machi.proto [hunk 2]
```diff
@@ -258,8 +272,9 @@ message Mpb_Request {
     optional Mpb_AppendChunkReq append_chunk = 112;
     optional Mpb_WriteChunkReq write_chunk = 113;
     optional Mpb_ReadChunkReq read_chunk = 114;
-    optional Mpb_ChecksumListReq checksum_list = 115;
-    optional Mpb_ListFilesReq list_files = 116;
+    optional Mpb_TrimChunkReq trim_chunk = 115;
```
Comment (slfritchie) [inline src/machi.proto:275]: In theory, these code assignments should be static.  But we don't have to worry about PB backward compatibility yet, so I'm not worried.

**Comment by slfritchie** [comment]
Kota: one small typo to fix, then +1, merge it, many thanks!

File: src/machi.proto (hunk 2 above)
Comment (kuenishi) [inline src/machi.proto:275]: Yes, at some point in future we should freeze them but now we're now still in early stage. That's why I took readability that XXXXChunkReq had better be in same continuous sequence, than compatibility.

**Comment by kuenishi** [comment]
YW and thanks!

**Pull request merged**